aspnetcore/blazor/lifecycle.md

```diff
@@ -266,3 +266,66 @@ For more information on the `RenderMode`, see <xref:blazor/hosting-model-configu
 ## Detect when the app is prerendering
 
 [!INCLUDE[](~/includes/blazor-prerendering.md)]
+
+## Cancelable background work
+
+Apps that perform long-running background work may need to cancel or suspend the background work. The following list indicates some of the reasons why background work items might require cancellation:
+
+* An executing background task was started with faulty input data or processing parameters.
+* The current set of executing background work items must be replaced with a new set of work items.
+* The priority of currently executing tasks must be changed.
+* The app has to be shut down in order to redeploy it to the server.
+* Server resources become limited, necessitating the rescheduling of backgound work items.
+
+To implement a cancelable background work pattern in a component:
+
+* Use a <xref:System.Threading.CancellationTokenSource> and <xref:System.Threading.CancellationToken>.
+* On [disposal of the component](#component-disposal-with-idisposable) and at any point cancellation is desired, call [CancellationTokenSource.Cancel](xref:System.Threading.CancellationTokenSource.Cancel%2A) to signal that the background work should be cancelled.
+* After the asynchronous call returns, call `_cts.Token.ThrowIfCancellationRequested();`.
+
+```razor
+@implements IDisposable
+@using System.Threading
+<button @onclick="LongRunningWork">Trigger long running work</button>
+
+@code {
+    private Resource _resource = new Resource();
+    private CancellationTokenSource _cts = new CancellationTokenSource();
+
+    protected async Task LongRunningWork()
+    {
+        await Task.Delay(5000, _cts.Token);
+        _cts.Token.ThrowIfCancellationRequested();
+        _resource.DoSomething();
+    }
+
+    public void Dispose()
+    {
+        _cts.Cancel();
+        _resource?.Dispose();
+    }
+
+    private class Resource : IDisposable
+    {
+        private bool _disposed;
+
+        public void Dispose()
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(nameof(Resource));
```

Comment:
It's usually fine to be resillient to multiple disposes. It would change this to no-op rather than throw.